src/silo/query_engine/actions/mutations.cpp

```diff
@@ -69,29 +69,88 @@ void Mutations<SymbolType>::addMutationsCountsForPosition(
    SymbolMap<SymbolType, std::vector<uint32_t>>& count_of_mutations_per_position
 ) {
    for (const auto& [filter, sequence_store_partition] : bitmaps_to_evaluate.bitmaps) {
-      for (const auto symbol : SymbolType::VALID_MUTATION_SYMBOLS) {
-         if (sequence_store_partition.positions[position].symbol_whose_bitmap_is_flipped != symbol) {
-            count_of_mutations_per_position[symbol][position] += filter->and_cardinality(
-               sequence_store_partition.positions[position].bitmaps.at(symbol)
+      for (const auto symbol : SymbolType::SYMBOLS) {
+         if (sequence_store_partition.positions[position].isSymbolDeleted(symbol)) {
+            continue;
+         }
+         if (sequence_store_partition.positions[position].isSymbolFlipped(symbol)) {
+            count_of_mutations_per_position[symbol][position] += filter->andnot_cardinality(
+               *sequence_store_partition.positions[position].getBitmap(symbol)
             );
          } else {
-            count_of_mutations_per_position[symbol][position] += filter->andnot_cardinality(
-               sequence_store_partition.positions[position].bitmaps.at(symbol)
+            count_of_mutations_per_position[symbol][position] += filter->and_cardinality(
+               *sequence_store_partition.positions[position].getBitmap(symbol)
             );
          }
       }
    }
    // For these partitions, we have full bitmaps. Do not need to bother with AND
    // cardinality
    for (const auto& [filter, sequence_store_partition] : bitmaps_to_evaluate.full_bitmaps) {
-      for (const auto symbol : SymbolType::VALID_MUTATION_SYMBOLS) {
-         if (sequence_store_partition.positions[position].symbol_whose_bitmap_is_flipped != symbol) {
+      for (const auto symbol : SymbolType::SYMBOLS) {
+         if (sequence_store_partition.positions[position].isSymbolDeleted(symbol)) {
+            continue;
+         }
+         if (sequence_store_partition.positions[position].isSymbolFlipped(symbol)) {
             count_of_mutations_per_position[symbol][position] +=
-               sequence_store_partition.positions[position].bitmaps.at(symbol).cardinality();
+               sequence_store_partition.sequence_count -
+               sequence_store_partition.positions[position].getBitmap(symbol)->cardinality();
          } else {
             count_of_mutations_per_position[symbol][position] +=
-               sequence_store_partition.sequence_count -
-               sequence_store_partition.positions[position].bitmaps.at(symbol).cardinality();
+               sequence_store_partition.positions[position].getBitmap(symbol)->cardinality();
+         }
+      }
+   }
+
+   correctMutationCountForDeletedSymbolBitmaps(
+      position, bitmaps_to_evaluate, count_of_mutations_per_position
+   );
+}
+
+template <typename SymbolType>
+// NOLINTNEXTLINE(readability-function-cognitive-complexity)
+void Mutations<SymbolType>::correctMutationCountForDeletedSymbolBitmaps(
```

Comment:
Why can't we incorporate those steps in the function above instead of the `if isSymbolDeleted { continue }`? IMO that would make the flow a little more obvious.

Reply:
I refactored it a little, which should make the flow more obvious